crates/analyzer/src/handlers/check_type.rs

```diff
@@ -210,12 +210,62 @@ impl VerylGrammarTrait for CheckType<'_> {
                                         ));
                                     }
                                 }
+                                GenericBoundKind::Inst(proto) => {
+                                    let proto_match = if arg.is_resolvable() {
```

Comment:
At Rust 2024 (Rust 1.85), if-let chain will be stabilized.
After it, this large if chain can be re-written like below. So `TODO` comment may be useful to mark for re-writing.

```rust
if arg.is_resolvable()
  && let Ok(symbol) = ...
  && let SymbolKind::Instance(x) = ...
  && let Ok(x) = ...
  && let Some(ref x) = x.found.proto() {
  ...
} else {
  false
}
```

Reply:
I added todo comments.

https://github.com/veryl-lang/veryl/blob/c353e6138556dd4d5c13571e37eb9b3cbce7700d/crates/analyzer/src/handlers/check_type.rs#L214
https://github.com/veryl-lang/veryl/blob/c353e6138556dd4d5c13571e37eb9b3cbce7700d/crates/analyzer/src/handlers/check_type.rs#L267